A little bit of repo overhaul

kurtbahartr opened the pull request:
#### Describe your PR, what does it fix/add?

It does a little overhaul throughout the configuration and adds some missing dependencies that are required on top of the most minimal Arch installation physically possible.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

I don't think there's anything extra for me to mention here.

#### Is it ready for merging, or does it need work?

It seems to be ready for merging, but more testing doesn't hurt. :)

kurtbahartr committed:
Install opensssh as a preliminary dependency too

kurtbahartr committed:
Install some missing packages that should be declared in pre_tasks

kurtbahartr committed:
Make Nvidia GPU preference more granular

Let's have the user decide whether to use the proprietary drivers
or not. The default of "false" is to make sure it all works fine
even on laptops and desktop computers with ancient Nvidia GPUs that
the latest proprietary drivers don't support.

See TODO in pre_tasks/gpu_drivers/all.yml as to what to do next.

kurtbahartr committed:
Make terms for package managers more generic

Diana seems to have plans to make this configuration more
cross-distribution where possible just like I thought and I guess this
should do just fine as first steps.

Test OK.

kurtbahartr committed:
Add configuration for new Nvidia cards

Will work to make it actually do what it's supposed to do later.

dianaw353 commented on pre_tasks/gpu_drivers/Archlinux.yml:
```diff
@@ -56,5 +56,5 @@
   become: true
   when:
-    - is_nvidia
+    - new_nvidia
```
Why is this new_nvidia when it should be is_nvidia

fkjbvfdvbnfdkjvbjfhd replied:
Because this task installs the mainstream proprietary drivers, which are meant for *new* Nvidia GPUs and not *all* of them  - Trying to run those on an old GPU would have it tell you that the GPU will be completely blacklisted from the system instead of attempting to probe it, also with a recommended version to install instead.

kurtbahartr committed:
Commonize some replicated packages for GPUs

This should also ensure compatibility with obscure GPU hardware that aren't
mainstream or are virtual and meant for VMs.

kurtbahartr committed:
Some good overhaul for Nvidia GPUs

I'm still not done yet and this will definitely haunt me in my dreams
tonight, but this should do just fine for now.

kurtbahartr committed:
CHANGELOG TIME!

kurtbahartr committed:
Move nvidia-utils from common to modern

Turns out nvidia-utils itself is also a part of the modern set and
actually hits into conflict with the one included in legacy drivers.
